tests/strategies/parse/test_image.py

```diff
@@ -70,37 +74,36 @@ def test_image(
         if failed:
             pytest.skip("Could not find Ghostscript on the system.")
 
-    mime_to_format = {"jpg": "jpeg", "jp2": "jpeg2000"}
-
     sample_file = static_files / filename
     reference_file = (
         static_files / filename_cropped if filename_cropped else static_files / filename
     )
     assert sample_file.exists(), f"Test file not found at {sample_file} !"
     assert reference_file.exists(), f"Test file not found at {reference_file} !"
 
-    for config_parse_test in range(2):
-        print(f"config_parse_test={config_parse_test}")
-
+    for itest in range(2):
         config = {
             "downloadUrl": sample_file.as_uri(),
             "mediaType": f"image/{image_format}",
-            "configuration": {"crop": crop} if crop and config_parse_test == 0 else {},
+            "configuration": {"crop": crop} if crop and itest == 0 else {},
         }
-        parser: "IParseStrategy" = ImageDataParseStrategy(parse_config=config)
-
-        session = parser.initialize(
-            {"imagecrop": crop} if crop and config_parse_test == 1 else None
-        )
-        session = parser.get(
-            {"imagecrop": crop} if crop and config_parse_test == 1 else session
-        )
-
-        assert (
-            session.get("cropped", False) if crop else not session.get("cropped", False)
-        )
-        assert (
-            session.get("format", SupportedFormat).value
-            == mime_to_format.get(image_format, image_format).upper()
-        )
-        assert session.get("content", b"") == reference_file.read_bytes()
+        session = {"imagecrop": crop} if crop and itest == 1 else {}
+
+        parser = ImageDataParseStrategy(parse_config=config)
+        session.update(parser.initialize(session))
+
+        parser = ImageDataParseStrategy(parse_config=config)
+        session.update(parser.get(session))
+
+        cache = DataCache()
+        image_key = session["image_key"]
+        data = cache.get(image_key)
```

Comment:
There should be a `del image_key` here somewhere as well I suppose?
Please put this in another `try/finally` block - essentially we want to make sure the values are always removed from the cache.